config/moved_config.go

```diff
@@ -147,9 +147,20 @@ func (cfg *Config) validateService() error {
 			}
 		}
 	}
+
 	return nil
 }
 
+// since mustValidateService will be used for next version, so use nolint to escape the `unused` lint error
+// nolint
+func (cfg *Config) mustValidateService() error {
```

Comment:
In Go, "Must*" functions typically mean that the call never fails (no errors), resulting in panic in case of an error. I'm not saying you should panic here, just that you should set the expectation straight by not using "must" as part of the name.

Reply:
That make senses. I originally tried to follow [example 2](https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#example-2---changing-the-return-values-of-a-function), so I name the new function to mustValidateService.

What name do you think is better, is `newValidateService` or `newValidateService` ok?  @jpkrohling